test/jaxp/javax/xml/jaxp/unittest/xpath/XPathExpFnTest.java

```diff
@@ -0,0 +1,222 @@
+                {"namespace-uri(//Customer)", ""},
+                {"namespace-uri(//foo:Customer)", "foo"},
+                {"namespace-uri(//@id)", ""},
+                {"namespace-uri(//@foo:id)", "foo"},
+                {"name(//*[namespace-uri()=\"foo\"])", "foo:Customer"},
+                {"name(//Customer)", "Customer"},
+                {"name(//foo:Customer)", "foo:Customer"},
+                {"name(//Customer/@id)", "id"},
+                {"name(//foo:Customer/@foo:id)", "foo:id"},
+                {"name(//*[name()='foo:Customer'])", "foo:Customer"},
+        };
+    }
+
+    /**
+     * This test evaluates XPath expressions of id function and checks against
+     * the expected result.
+     *
+     * @param exp      XPath expression
+     * @param expected expected result
+     * @throws Exception
+     */
+    @Test(dataProvider = "idExpTestCases")
+    void testIdFn(String exp, String expected) throws Exception {
+        XPath xPath = XPathFactory.newInstance().newXPath();
+
+        Node node = xPath.evaluateExpression(exp, doc, Node.class);
+        Node node2 = (Node) xPath.evaluate(exp, doc, XPathConstants.NODE);
+
+        Assert.assertEquals(node.getNodeName() + "_" +
+                        node.getAttributes().item(0).getNodeValue()
+                , expected);
+        Assert.assertEquals(node2, node);
+    }
+
+    /**
+     * This test evaluates XPath expressions of count function and checks
+     * against the expected result.
+     *
+     * @param exp      XPath expression
+     * @param expected expected result
+     * @throws Exception
+     */
+    @Test(dataProvider = "countExpTestCases")
+    void testCountFn(String exp, int expected) throws Exception {
+        XPath xPath = XPathFactory.newInstance().newXPath();
+
+        double num = xPath.evaluateExpression(exp, doc, Double.class);
+        double num2 = (double) xPath.evaluate(exp, doc, XPathConstants.NUMBER);
+
+        Assert.assertEquals(num, expected);
+        Assert.assertEquals(num2, num);
+    }
+
+    /**
+     * This test evaluates XPath expressions of position function and checks
+     * against the expected result.
+     *
+     * @param exp      XPath expression
+     * @param expected expected result
+     * @throws Exception
+     */
+    @Test(dataProvider = "positionExpTestCases")
+    void testPositionFn(String exp, String expected) throws Exception {
+        XPath xPath = XPathFactory.newInstance().newXPath();
+
+        Node node = xPath.evaluateExpression(exp, doc, Node.class);
+        Node node2 = (Node) xPath.evaluate(exp, doc, XPathConstants.NODE);
+
+        Assert.assertEquals(node.getNodeName() + "_" +
+                        node.getAttributes().item(0).getNodeValue()
+                , expected);
+        Assert.assertEquals(node2, node);
+    }
+
+    /**
+     * This test evaluates XPath expressions of name and local-name functions
+     * and checks against the expected result.
+     *
```

Comment:
Verifies that the result of evaluating the name and local-name functions matches the expected result.